### ISSUE TITLE
[feature request] List item close

### ISSUE THREAD

**Issue by tmrobert8** [NEEOInc/neeo-sdk#111]
As I'm implementing some list items - I really would like the ability to return something from the action that specifies whether to close the list or not.  I don't really like that when you click on a leaf item (which invokes an action) that you then have hit the back button to get back to the 'pre' list screen (seems like an added step that is not very user friendly - if you are clicking on an item, you are taking action and that 'action' should close when completed).

Seems like that would be the 'major' use case.  A 'minor' use case (which is what you have implemented) would be to have the list stay open after clicking to click on something else.  

I'd think a majority of the time - you execute the action and are done.

Example:
Take a look at @nklerk KODI example.  When you select a movie to watch - what's the next action you are likely to do?  I'm betting it's not select another movie - it's probably start using the controls (volume, transport, etc).

I do see some value in the 'minor' use case however - so I really think it should be up to the plugin writer to determine which case is better suited for their plugin. I think the action callback should return a boolean (or an action type of object like ```{ "action" : "close" } ``` for most flexibility) that would specify what to do

**Comment by neophob** [comment]
Fully agree, thats on the roadmap

**Comment by tmrobert8** [comment]
Nice - btw this works great as a list selector as well.  Example: I have a 'source' on my Russound whole house where "1" is "AM/FM", "2" is "Pandora", etc.  Before this, I used a slider to specify the source (which looks bad, is non-intuitive and couldn't be labelled).  However, I just converted that to the new 'directory' type - now I can push the "Source" button and select "Pandora" from the list and have it switch to it.

Not exactly a 'directory' but is VERY useful feature...

**Comment by neophob** [comment]
Yes - we were talking about that internally - and favorites and directories do in fact share quite a lot.
